storage/src/main/scala/coop/rchain/storage/regex/RegexPattern.scala

```diff
@@ -170,8 +195,172 @@ object CharClassPattern extends ParsedPattern {
   def apply(charSet: Set[Char], negateCharSet: Boolean): CharClassPattern =
     new CharClassPattern(charSet, negateCharSet)
 
-  def tryParse(str: CharSequence): Option[(CharClassPattern, Int)] =
-    throw new NotImplementedError("TODO")
+  private[this] val escapes =
+    Map('t' -> '\t', 'r' -> '\r', 'n' -> '\n', 'f' -> '\f', 'v' -> '\u0011')
+
+  private[this] val allSpecialChars = "\\\\[]|().?*+{}".toSet
+
+  //region predefined char classes
+  private[this] val wordCharClass: CharClassPattern = CharClassPattern(
+    "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz")
+
+  private[this] val nonWordCharClass: CharClassPattern = CharClassPattern(
+    "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz",
+    negateCharSet = true)
+
+  private[this] val digitsCharClass: CharClassPattern = CharClassPattern("0123456789")
+
+  private[this] val nonDigitsCharClass: CharClassPattern =
+    CharClassPattern("0123456789", negateCharSet = true)
+
+  private[this] val spacesCharClass: CharClassPattern = CharClassPattern("\t\n\11\f\r ")
```

Comment:
Do we care about unicode spaces? 
http://jkorpela.fi/chars/spaces.html
See also http://www.fileformat.info/info/unicode/char/ffa0/index.htm

Reply:
I've added well-known unicode spaces to the set. However that's something we might need to think twice about: Java and PHP engines assumes that \s is only ASCII spaces, and for unicode spaces someone should use {Zs} category, in contrast - most of other regex engines (Javascript, .NET, Go, etc.) - assumes that \s matches all unicode space chars. From the other side - primary purpose of our regex engine (as I understand) - is uri-like key access and unification for rchain storage, not a general purpose Regex engine. If so - it's common for Uri engines to deny use of Unicode spaces (and other special chars) to reduce phishing risks.